### ISSUE TITLE
StrictMode - Not accepting dial code change while typing

### ISSUE THREAD

**Issue by DanyDes** [jackocnr/intl-tel-input#1777]
### Plugin version
v24.3.6

### Steps to reproduce
1. Init the property `strictMode: true` 
2. Fill in just the national number, in my case 79554053
3.  Go to the beginning and add the dialing code, in my case +591 "Bolivia"
4.  The dial code selector gets the value +597 "Suriname" and I can't type anymore

### Expected behaviour
While typing at the beginning of the number it should let me complete the dial code of the country +591(number)
Result: +59179554053
### Actual behaviour
While typing it stops at +59(number)
Result: +5979554053

### Initialisation options
```autoPlaceholder: "off",
initialCountry: "auto",
nationalMode: false,
autoHideDialCode: false,
strictMode: true
utilsScript: '/build/intl-tel-input/js/utils.js',
```

Why do I have to do that? because users used to write national phone numbers, but now the platform has accepted international users and local people will update their phone numbers.

**Comment by jackocnr** [comment]
Thanks for raising this. I've changed it so that it ignores the max length if the cursor is not at the end, to make sure things like this don't happen. Fix released in v24.3.7.

**Comment by DanyDes** [comment]
Thank you
